extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsStorageDruidModule.java

```diff
@@ -126,5 +130,23 @@ public void configure(Binder binder)
     Binders.taskLogsBinder(binder).addBinding("hdfs").to(HdfsTaskLogs.class);
     JsonConfigProvider.bind(binder, "druid.indexer.logs", HdfsTaskLogsConfig.class);
     binder.bind(HdfsTaskLogs.class).in(LazySingleton.class);
+
+    final String principal = props.getProperty("druid.hdfs.security.kerberos.principal");
+    final String keytab = props.getProperty("druid.hdfs.security.kerberos.keytab");
+    if (!Strings.isNullOrEmpty(principal) && !Strings.isNullOrEmpty(keytab)) {
```

Comment:
ah, I guess the problem is neither druid-hadoop-indexing nor druid-hdfs-storage depend on each other, and there's no druid-hadoop-common. ok, I think the code duplication is fine then, it's probably not worth creating a new common module. Perhaps add a comment to each location that the same code is in the other location – this'll help maintainers remember to fix both locations if anyone ever needs to fix a bug.